paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/migrate/HiveMigrator.java

```diff
@@ -0,0 +1,334 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import static org.apache.paimon.hive.HiveTypeUtils.toPaimonType;
+
+/** Migrate hive table to paimon table. */
+public class HiveMigrator implements Migrator {
+
+    private static final Predicate<FileStatus> HIDDEN_PATH_FILTER =
+            p -> !p.getPath().getName().startsWith("_") && !p.getPath().getName().startsWith(".");
+
+    private final FileIO fileIO;
+    private final HiveCatalog hiveCatalog;
+    private final IMetaStoreClient client;
+    private final String sourceDatabase;
+    private final String sourceTable;
+    private final String targetDatabase;
+    private final String targetTable;
+    private final Map<String, String> options;
+
+    public HiveMigrator(
+            HiveCatalog hiveCatalog,
+            String sourceDatabase,
+            String sourceTable,
+            String targetDatabase,
+            String targetTable,
+            Map<String, String> options) {
+        this.hiveCatalog = hiveCatalog;
+        this.fileIO = hiveCatalog.fileIO();
+        this.client = hiveCatalog.getHmsClient();
+        this.sourceDatabase = sourceDatabase;
+        this.sourceTable = sourceTable;
+        this.targetDatabase = targetDatabase;
+        this.targetTable = targetTable;
+        this.options = options;
+    }
+
+    public void executeMigrate(boolean sync) throws Exception {
+        if (!client.tableExists(sourceDatabase, sourceTable)) {
+            throw new RuntimeException("Source hive table does not exist");
+        }
+
+        Table sourceHiveTable = client.getTable(sourceDatabase, sourceTable);
+        Map<String, String> properties = new HashMap<>(sourceHiveTable.getParameters());
+        checkPrimaryKey();
+
+        AbstractFileStoreTable paimonTable =
+                createPaimonTableIfNotExists(
+                        client.getSchema(sourceDatabase, sourceTable),
+                        sourceHiveTable.getPartitionKeys(),
+                        properties);
+        checkPaimonTable(paimonTable);
+
+        List<String> partitionsNames =
+                client.listPartitionNames(sourceDatabase, sourceTable, Short.MAX_VALUE);
+        checkCompatible(sourceHiveTable, paimonTable);
+
+        List<MigrateTask> tasks = new ArrayList<>();
+        if (partitionsNames.isEmpty()) {
+            tasks.add(importUnPartitionedTableTask(fileIO, sourceHiveTable, paimonTable));
+        } else {
+            tasks.addAll(
+                    importPartitionedTableTask(
+                            client, fileIO, partitionsNames, sourceHiveTable, paimonTable));
+        }
+
+        if (sync) {
```

Comment:
Why we need a sync mode? Actually, single thread mode.

Reply:
Reasonable, we don't need single thread mode. At first, I want to return a Future<MigrateResult>, but seems it is not necessary.